### DISCUSSION TITLE
Use data-analytics tags without double quotes? (for Webflow)

### DISCUSSION THREAD

**Discussion by dschuermann** [plausible/analytics#2284] [Anything Else]
I was using Webflow with plausible's data-analytics tags as described here: https://plausible.io/docs/custom-event-goals#1-add-a-data-analytics-attribute-tag-on-the-link-you-want-to-track

It worked really well and is a great way to keep custom goals organized. It worked even though Webflow generated invalid HTML with double quotes inside double quotes:

![Screenshot_20220929_163338](https://user-images.githubusercontent.com/321888/193060391-c3f68907-45da-44d1-9bf3-4085794c0f27.png)

Today, Webflow changed something and now I am not able to insert double quotes inside these custom attributes:

![Screenshot_20220929_161803](https://user-images.githubusercontent.com/321888/193059724-f8902e17-2dee-4c7e-8039-251603c632b9.png)

Anyone know a way around this? How can I still keep using data-analytics tags?

**Comment by metmarkosaric** [comment]
thanks for reporting that @dschuermann! sounds weird. could you please ask webflow team if they made a change or what they recommend now? it would be nice if they can share what's going on.

from our side, we do have some code examples that should work for webflow too here: https://plausible.io/docs/custom-event-goals#code-examples-for-website-builders-such-as-webflow-shopify-and-wix. we're also working on an easier way to setup custom events on the wbesite builder tools which should help as well. we'll post the details as soon as it's ready

**Reply by dschuermann** [reply]
Hey @metmarkosaric , thanks for answering.
The anchor of your link doesn't exist on the page. Are you recommending to use javascript calls for each button in Webflow?

Is there any way I can still use data-analytics custom attributes but with single quotes inside double quotes?

like this?:
![Screenshot_20220929_171615](https://user-images.githubusercontent.com/321888/193070987-625c9767-b715-423e-ad63-a31871184f4e.png)

**Reply by dschuermann** [reply]
@metmarkosaric The link now works, thanks. Any idea how we can use custom props for plausible here?

**Reply by RobertJoonas** [reply]
Hi @dschuermann!

To use custom properties you would have to modify the JavaScript code and read the properties and values from the HTML element attributes. 

We're currently working on a new script extension that allows tracking links (and other HTML elements) through custom attributes. This will make it much simpler (no mess with the quotes) and you won't need to insert any JavaScript yourself. The new extension will easily allow using custom properties too.

Hopefully we'll be able to ship it by the end of the month. Thanks for your patience!

**Reply by dschuermann** [reply]
@RobertJoonas awesome, thanks. Really looking forward to it.

**Reply by metmarkosaric** [reply]
hi @dschuermann! we now have a new guide for setting up custom events on Webflow if you're still looking to set this up: https://plausible.io/docs/webflow-integration

**Reply by dschuermann** [reply]
@metmarkosaric Looks great, we'll probably migrate to it at some point in the next weeks, thanks!

**Comment by Alexander01998** [comment]
Here is an improved version of the [data-analytics script](https://web.archive.org/web/20221202123219/https://plausible.io/docs/custom-event-goals#2-add-the-javascript-that-will-be-sending-the-link-click-events-to-plausible) that doesn't need double-quotes:

```javascript
<script>
  let links = document.querySelectorAll("a[data-analytics]");
  for (var i = 0; i < links.length; i++) {
    links[i].addEventListener('click', handleLinkEvent);
    links[i].addEventListener('auxclick', handleLinkEvent);
  }
  function handleLinkEvent(event) {
    var link = event.target;
    var middle = event.type == "auxclick" && event.which == 2;
    var click = event.type == "click";
    while (link && (typeof link.tagName == 'undefined' || link.tagName.toLowerCase() != 'a' || !link.href)) {
      link = link.parentNode;
    }
    if (middle || click) {
      let eventName = link.getAttribute('data-analytics');
      let eventProps = {};
      let eventPropsAttrs = link.attributes;
      for (var i = 0; i < eventPropsAttrs.length; i++) {
        if (eventPropsAttrs[i].name.startsWith("data-analytics-")) {
          let propName = eventPropsAttrs[i].name.replace("data-analytics-", "");
          eventProps[propName] = eventPropsAttrs[i].value;
        }
      }
      plausible(eventName, { props: eventProps });
    }
    if (!link.target) {
      if (!(event.ctrlKey || event.metaKey || event.shiftKey) && click) {
        setTimeout(function () {
          location.href = link.href;
        }, 150);
        event.preventDefault();
      }
    }
  }
</script>
```

To use this modified script, write your tagged links like this:

```html
<a href="..." data-analytics="Download" data-analytics-method="HTTP" data-analytics-region="Europe">some link</a>
```

Note, however, that the [Plausible docs](https://plausible.io/docs/custom-event-goals) no longer mention data-analytics at all. It looks to me like we are now supposed to use the tagged-events extension instead (which is unfortunate because I prefer using data attributes instead of CSS classes).

@metmarkosaric I would love to hear your thoughts on this. Would you advise against using data-analytics now that the tagged-events extension exists, or has it simply been deleted from the docs because not many people were using it? What do you think of my modification to make data-analytics easier to use?

**Reply by metmarkosaric** [reply]
thanks @Alexander01998! the previous method still works and is fully supported. we haven't made any changes to it and there's still a section in the docs that explains how to trigger custom events manually with javascript. we just made it simpler to set up custom events directly in the UI of many CMS tools such as WordPress by supporting CSS classes. our previous method with javascript was difficult to use for many people. but if you know how and prefer the previous method, it still works all fine and there are no drawbacks compared to the new method

**Comment by dschuermann** [comment]
Just FYI and as a feedback: We finally migrated to use tagged-events as described on https://plausible.io/docs/webflow-integration

We added the mapping from element IDs to goals in the site's custom code to have them at the mapping in one place (as described in step 3, https://plausible.io/docs/webflow-integration#3-trigger-custom-events-with-javascript-on-your-site)

The proposed script assumes that these element IDs are now on all pages, which is not the case, so we altered the script to check that the element is actually present on the page:
```
document.addEventListener('DOMContentLoaded', function (_e) {
    toTag.forEach(function (tagObject) {
        var element = document.getElementById(tagObject.elementId)
        tagObject.classes.split(' ').forEach(function (className) {
            if (element) {
                element.classList.add(className)
            }
        })
    })
})
```

You can also check out the source on https://www.heylogin.com/en to see this.

**Reply by metmarkosaric** [reply]
thanks for sharing @dschuermann! appreciate it and happy it's all working for you now!

**Reply by dschuermann** [reply]
@metmarkosaric I would propose to also change the script in the docs. Currently it really only works if _all_ pages of a site always have _all_ goals.

**Reply by metmarkosaric** [reply]
for sure! it's a useful change and we've updated the docs now! thanks again!
